com.unity.multiplayer.mlapi/Runtime/Core/BatchingUtil.cs

```diff
@@ -0,0 +1,187 @@
+                {
+                    SendDict[clientId] = new SendStream();
+                    SendDict[clientId].Item = item;
+                    SendDict[clientId].Writer = new PooledBitWriter(SendDict[clientId].Stream);
+
+                    SendDict[clientId].Writer.WriteBit(false); // Encrypted
+                    SendDict[clientId].Writer.WriteBit(false); // Authenticated
+
+                    switch (item.QueueItemType)
+                    {
+                        case RPCQueueManager.QueueItemType.ServerRPC:
+                            SendDict[clientId].Writer.WriteBits(MLAPIConstants.MLAPI_STD_SERVER_RPC, 6); // MessageType
+                            break;
+                        case RPCQueueManager.QueueItemType.ClientRPC:
+                            SendDict[clientId].Writer.WriteBits(MLAPIConstants.MLAPI_STD_CLIENT_RPC, 6); // MessageType
+                            break;
+                    }
+                }
+
+                // write the amounts of bytes that are coming up
+                PushLength(item.MessageData.Count, ref SendDict[clientId].Writer);
+
+                // write the message to send
+                // todo: is there a faster alternative to .ToArray()
+                SendDict[clientId].Writer.WriteBytes(item.MessageData.ToArray(), item.MessageData.Count);
+
+                ProfilerStatManager.bytesSent.Record((int)item.MessageData.Count);
+                ProfilerStatManager.rpcsSent.Record();
+            }
+        }
+
+        public delegate void SendCallbackType(ulong clientId, SendStream messageStream);
+        public delegate void ReceiveCallbackType(BitStream messageStream, MLAPI.RPCQueueManager.QueueItemType messageType, ulong clientId, float time);
+
+        /// <summary>
+        /// SendItems
+        /// Send any batch of RPC that are of length above threshold
+        /// </summary>
+        /// <param name="threshold"> the threshold in bytes</param>
+        /// <param name="sendCallback"> the function to call for sending the batch</param>
+        public void SendItems(int threshold, SendCallbackType sendCallback)
+        {
+            List<ulong> sent = new List<ulong>();
+
+            foreach (KeyValuePair<ulong, SendStream> entry in SendDict)
+            {
+                // read the queued message
+                using PooledBitWriter writer = SendDict[entry.Key].Writer;
+                int length = (int)writer.GetStream().Length;
+
+                if (length >= threshold)
+                {
+                    sendCallback(entry.Key, entry.Value);
+                    ProfilerStatManager.rpcBatchesSent.Record();
+
+                    // mark the client for which a batch was sent
+                    sent.Add(entry.Key);
+                }
+            }
+
+            // clear the batch that were sent from the SendDict
+            foreach(ulong clientid in sent)
+            {
+                SendDict.Remove(clientid);
```

Comment:
I assume we can't just call this right after line 143 and avoid the 'sent' list?  Worth a comment for other folks like me that might hastily make this change and not know why it has to be this way

Reply:
Commented.